### DISCUSSION TITLE
inventory lock

### DISCUSSION THREAD

**Discussion by spitfire2112** [isid0re/SoloLeveling#111] [Q&A] [answered]
Where can I edit the inventory lock. I have a couple charms I want my character to keep while running but the inventory lock in the usual spot doesn't seem to work.

Thanks

**Comment by isid0re** [comment, marked as answer]
Currently the inventory lock wont work without removing the inventory check in town.clearjunk().

One of the project todos is having the bot pick and equip charms. I'm not there yet, but it is a goal.

**Reply by spitfire2112** [reply]
Understood.

Thank you for the quick response.

**Comment by zylo-** [comment]
I figured it out! The bot kept selling my anni and torches but if you just add this:
`		
                        items[i].classid !== 603 && // Annihilus
			items[i].classid !== 604 &&// Hellfire Torch

`

to SoloLeveling\Functions\TownOverrides.js to line 910 it will prevent the bot from selling your 2 most important charms.

**Reply by zylo-** [reply]
and

items[i].classid !== 605 && // Gheeds

for gheeds.

**Reply by spitfire2112** [reply]
Hell yeah, my dude! Thanks for figuring that out. 
I'm assuming you've tested this, and it's proven to work?

Only ask because I've already lost a torch, lol

**Reply by zylo-** [reply]
Yeah I lost 2x torch anni gheeds until I figured this out! It 100% works. My characters have been going for over 12 hours now and they still have torch + anni in their inventories! The bot moves them around a lot though in the inventory.

**Reply by spitfire2112** [reply]
Nice! And yeah, that's slightly annoying from an OCD perspective, haha, but I'm not going to complain.

Thanks for getting this worked out. I appreciate it.

**Reply by isid0re** [reply]
This is a good catch for preventing item selling. I will add it in. While it doesnt fix the item lock it's a great start. Thanks.

**Reply by ddarbs** [reply]
hmm I added these to 910, and it isn't selling them but it is banking them. Any thoughts?

**Reply by spitfire2112** [reply]
> I figured it out! The bot kept selling my anni and torches but if you just add this:
> `
> items[i].classid !== 603 && // Annihilus
> items[i].classid !== 604 &&// Hellfire Torch
> 
> `
> 
> to SoloLeveling\Functions\TownOverrides.js to line 910 it will prevent the bot from selling your 2 most important charms.

Will you please copy and paste your line 910 to make sure I'm doing this properly?

**Reply by ddarbs** [reply]
908-936 thank you!
for (i = 0; !!items && i < items.length; i += 1) {
		if ([18, 41, 76, 77, 78].indexOf(items[i].itemType) === -1 && // Don't drop tomes, keys or potions
			items[i].classid !== 603 && // Annihilus
			items[i].classid !== 604 && // Hellfire Torch
			items[i].classid !== 605 && // Gheeds
			items[i].classid !== 524 && // Scroll of Inifuss
			items[i].classid !== 525 && // Key to Cairn Stones
			items[i].classid !== 549 && // Horadric Cube
			items[i].classid !== 92 && // Staff of Kings
			items[i].classid !== 521 && // Viper Amulet
			items[i].classid !== 91 && // Horadric Staff
			items[i].classid !== 552 && // Book of Skill
			items[i].classid !== 545 && // Potion of Life
			items[i].classid !== 546 && // A Jade Figurine
			items[i].classid !== 547 && // The Golden Bird
			items[i].classid !== 548 && // Lam Esen's Tome
			items[i].classid !== 553 && // Khalim's Eye
			items[i].classid !== 554 && // Khalim's Heart
			items[i].classid !== 555 && // Khalim's Brain
			items[i].classid !== 173 && // Khalim's Flail
			items[i].classid !== 174 && // Khalim's Will
			items[i].classid !== 644 && // Malah's Potion
			items[i].classid !== 646 && // Scroll of Resistance
			(items[i].code !== 529 || !!me.findItem(518, 0, 3)) && // Don't throw scrolls if no tome is found (obsolete code?)
			(items[i].code !== 530 || !!me.findItem(519, 0, 3)) && // Don't throw scrolls if no tome is found (obsolete code?)
			!Cubing.keepItem(items[i]) && // Don't throw cubing ingredients
			!Runewords.keepItem(items[i]) && // Don't throw runeword ingredients
			!CraftingSystem.keepItem(items[i]) // Don't throw crafting system ingredients
		)

**Comment by spitfire2112** [comment]
Awesome!!

Thanks

**Reply by isid0re** [reply]
The safety check has been added to the townoverrides.js
